presto-native-execution/presto_cpp/external/jwt-cpp/traits/nlohmann-json/traits.h

```diff
@@ -0,0 +1,135 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef JWT_CPP_NLOHMANN_JSON_TRAITS_H
+#define JWT_CPP_NLOHMANN_JSON_TRAITS_H
+
+/*
+  Note - changes to the original version for prestissimo:
+  - the path to jwt.h was updated
+  - the JSON header inclusion was changed to the local nlohman-json
+*/
+#include "presto_cpp/external/json/json.hpp"
+#include "presto_cpp/external/jwt-cpp/jwt.h"
+
+namespace jwt {
+namespace traits {
+struct nlohmann_json {
+  using json = nlohmann::json;
+  using value_type = json;
+  using object_type = json::object_t;
+  using array_type = json::array_t;
+  using string_type =
+      std::string; // current limitation of traits implementation
+  using number_type = json::number_float_t;
+  using integer_type = json::number_integer_t;
+  using boolean_type = json::boolean_t;
+
+  static jwt::json::type get_type(const json& val) {
+    using jwt::json::type;
+
+    if (val.type() == json::value_t::boolean)
+      return type::boolean;
+    // nlohmann internally tracks two types of integers
+    if (val.type() == json::value_t::number_integer)
+      return type::integer;
+    if (val.type() == json::value_t::number_unsigned)
+      return type::integer;
+    if (val.type() == json::value_t::number_float)
+      return type::number;
+    if (val.type() == json::value_t::string)
+      return type::string;
+    if (val.type() == json::value_t::array)
+      return type::array;
+    if (val.type() == json::value_t::object)
+      return type::object;
+
+    throw std::logic_error("invalid type");
+  }
+
+  static json::object_t as_object(const json& val) {
+    if (val.type() != json::value_t::object)
+      throw std::bad_cast();
+    return val.get<json::object_t>();
+  }
+
+  static std::string as_string(const json& val) {
+    if (val.type() != json::value_t::string)
+      throw std::bad_cast();
+    return val.get<std::string>();
+  }
+
+  static json::array_t as_array(const json& val) {
+    if (val.type() != json::value_t::array)
+      throw std::bad_cast();
+    return val.get<json::array_t>();
+  }
+
+  static int64_t as_int(const json& val) {
+    switch (val.type()) {
+      case json::value_t::number_integer:
+      case json::value_t::number_unsigned:
+        return val.get<int64_t>();
+      default:
+        throw std::bad_cast();
+    }
+  }
+
+  static bool as_bool(const json& val) {
+    if (val.type() != json::value_t::boolean)
+      throw std::bad_cast();
+    return val.get<bool>();
+  }
+
+  static double as_number(const json& val) {
+    if (val.type() != json::value_t::number_float)
+      throw std::bad_cast();
+    return val.get<double>();
+  }
+
+  static bool parse(json& val, std::string str) {
```

Comment:
Should str parameter be a const& ?

Reply:
Right, could be. Would save one copy. Again, this is as is from the jwt-cpp project. We can change it if needed. How much do we want to change it fro the original?